pyiceberg/table/__init__.py

```diff
@@ -2271,3 +2317,244 @@ def commit(self) -> Snapshot:
+                )
+            else:
+                raise ValueError(f"Cannot add duplicate partition field name: {existing_partition_field.name}")
+
+        self._name_to_added_field[new_field.name] = new_field
+        self._adds.append(new_field)
+        return self
+
+    def add_identity(self, source_column_name: str) -> UpdateSpec:
+        return self.add_field(self._unassigned_field_name, source_column_name, IdentityTransform())
+
+    def remove_field(self, name: str) -> UpdateSpec:
+        added = self._name_to_added_field.get(name)
+        if added:
+            raise ValueError(f"Cannot delete newly added field {name}")
+        renamed = self._renames.get(name)
+        if renamed:
+            raise ValueError(f"Cannot rename and delete field {name}")
+        field = self._name_to_field.get(name)
+        if not field:
+            raise ValueError(f"No such partition field: {name}")
+
+        self._deletes.add(field.field_id)
+        return self
+
+    def rename_field(self, name: str, new_name: str) -> UpdateSpec:
+        existing_field = self._name_to_field.get(new_name)
+        if existing_field and isinstance(existing_field.transform, VoidTransform):
+            return self.rename_field(name, name + "_" + str(existing_field.field_id))
+        added = self._name_to_added_field.get(name)
+        if added:
+            raise ValueError("Cannot rename recently added partitions")
+        field = self._name_to_field.get(name)
+        if not field:
+            raise ValueError(f"Cannot find partition field {name}")
+        if field.field_id in self._deletes:
+            raise ValueError(f"Cannot delete and rename partition field {name}")
+        self._renames[name] = new_name
+        return self
+
+    def commit(self) -> None:
+        new_spec = self._apply()
+        updates = []
+        requirements = []
+        if self._table.metadata.default_spec_id != new_spec.spec_id:
+            if new_spec.spec_id not in self._table.specs():
+                spec_update = AddPartitionSpecUpdate(spec=new_spec)
+                updates.append(spec_update)
+                if len(self._table.specs()) == 1:
+                    required_last_assigned_partitioned_id = PARTITION_FIELD_ID_START - 1
+                else:
+                    required_last_assigned_partitioned_id = self._table.spec().last_assigned_field_id
+                requirements.append(
+                    AssertLastAssignedPartitionId(last_assigned_partition_id=required_last_assigned_partitioned_id)
+                )
+            if self._transaction:
+                self._transaction._append_updates(*updates)  # pylint: disable=W0212
+                self._transaction._append_requirements(*requirements)  # pylint: disable=W0212
+            else:
+                updates.append(SetDefaultSpecUpdate(spec_id=new_spec.spec_id))
+                requirements.append(AssertDefaultSpecId(default_spec_id=self._table.metadata.default_spec_id))
```

Comment:
The metadata should probably have be marked as private 😁 
```suggestion
                requirements.append(AssertDefaultSpecId(default_spec_id=self._table.spec().spec_id))
```